tasks/aws-peer.yml

```diff
@@ -11,7 +11,7 @@
     state: present
     tags: "{{ item_peer.tags }}"
   register: ret_vpc_peer
-  when: "ansible_version.full | version_compare('2.5', '<')"
+  when: "ansible_version.full is version('2.5', '<')"
```

Comment:
Im not sure this works because ansible first tries to interpret the file (syntactic analysis) before running

Reply:
yes, it works. Ansible 2.9 deprecated the `version_compare` filter.

```

TASK [cloud-vpc.chaordic : AWS | PEER create using adapted module] *******************************************************
task path: /Users/marco/linx/dev/corp-cloudops-devops/ansible/roles/cloud-vpc.chaordic/tasks/aws-peer.yml:4
Monday 27 July 2020  15:35:14 -0300 (0:00:00.105)       0:00:09.584 *********** 
fatal: [localhost]: FAILED! => {
    "msg": "The conditional check 'ansible_version.full | version_compare('2.5', '<')' failed. The error was: template error while templating string: no filter named 'version_compare'. String: {% if ansible_version.full | version_compare('2.5', '<') %} True {% else %} False {% endif %}\n\nThe error appears to be in '/Users/marco/linx/dev/corp-cloudops-devops/ansible/roles/cloud-vpc.chaordic/tasks/aws-peer.yml': line 4, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n# Use custom module, latest, module when running older (<2.5) ansible\n- name: AWS | PEER create using adapted module\n  ^ here\n"
}

PLAY RECAP ***************************************************************************************************************
localhost                  : ok=5    changed=1    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```